url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/net/http/task/server/HttpServerBootTask.java

```diff
@@ -0,0 +1,29 @@
+package net.hetimatan.net.http.task.server;
+
+
+import java.lang.ref.WeakReference;
+
+import net.hetimatan.net.http.HttpServer;
+import net.hetimatan.util.event.EventTask;
+import net.hetimatan.util.event.EventTaskRunner;
+
+public class HttpServerBootTask extends EventTask {
+	public static final String TAG = "HttpServerBootTask";
+	private WeakReference<HttpServer> mServer = null;
+
+	public HttpServerBootTask(HttpServer httpServer) {
+		mServer = new WeakReference<HttpServer>(httpServer);
+		errorAction(new HttpServerClose(httpServer));
+	}
+
+	@Override
+	public String toString() {
+		return TAG;
+	}
+
+	@Override
+	public void action(EventTaskRunner runner) throws Throwable {
+		HttpServer server = mServer.get();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *server*. Execution paths that do not contain closure statements were detected.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).